nifi-nar-bundles/nifi-simulator-bundle/nifi-simulator-processors/src/main/java/com/apache/nifi/processors/simulator/GenerateTimeSeriesFlowFile.java

```diff
@@ -0,0 +1,180 @@
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor PRINT_HEADER = new PropertyDescriptor
+            .Builder().name("PRINT_HEADER")
+            .displayName("Print Header")
+            .description("Directs the processor whether to print a header line or not.")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+    public static final Relationship SUCCESS = new Relationship.Builder()
+            .name("Success")
+            .description("When the flowfile is successfully generated")
+            .build();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private Set<Relationship> relationships;
+
+    @Override
+    protected void init(final ProcessorInitializationContext context) {
+
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+
+        descriptors.add(SIMULATOR_CONFIG);
+        descriptors.add(PRINT_HEADER);
+
+        this.descriptors = Collections.unmodifiableList(descriptors);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(SUCCESS);
+        this.relationships = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public void onPropertyModified(final PropertyDescriptor descriptor, final String oldValue, final String newValue) {
+        if (SIMULATOR_CONFIG.equals(descriptor))
+            simConfig = null;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        loadConfiguration(context.getProperty(SIMULATOR_CONFIG).getValue());
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        ComponentLog logger = getLogger();
+        FlowFile flowFile = session.get();
+
+        // Create the flowfile, as it probably does not exist
+        if (flowFile == null)
+            flowFile = session.create();
+
+        // Get the data
+        String data = generateData(context.getProperty(PRINT_HEADER).asBoolean());
+
+        // Write the results back out to flow file
+        try{
+            flowFile = session.write(flowFile, out -> out.write(data.getBytes()));
+            session.getProvenanceReporter().create(flowFile);
+            session.transfer(flowFile, SUCCESS);
+        } catch (ProcessException ex) {
+            logger.error("Unable to write generated data out to flowfile. Error: ", ex);
+        }
+    }
+
+    // Loads the configuration from the file
+    private void loadConfiguration(String fileName){
```

Comment:
I would recommend doing away with this method as well and just doing the minimal loading logic in onScheduled.

Reply:
Will Do